[JDBC] - storeItemValueProvider does not store Time for DATETIMEITEM

Ubuntu 17.04, java-8-openjdk-amd64, Openhab 2.1, JDBC Persistence 1.10.

When persisting a DATETIMEITEM, only the date (yyyy-mm-dd) is set.  It seems that in JdbcBaseDAO, storeItemValueProvider(), that the java.sql.Date does not return time.  Modifying to store javaType java.sql.Timestamp seems to resolve this issue:

        } else if ("DATETIMEITEM".equals(itemType)) {
            // vo.setValueTypes(getSqlTypes().get(itemType), java.util.Date.class);
            vo.setValueTypes(getSqlTypes().get(itemType), java.sql.Timestamp.class);
            Calendar x = ((DateTimeType) item.getState()).getCalendar();
            java.sql.Timestamp d = new java.sql.Timestamp(x.getTimeInMillis());
            logger.debug("JDBC::storeItemValueProvider: DateTimeItem: '{}'", d);
            vo.setValue(d);

Debug Log of Issue:
2017-07-17 21:36:56.681 [DEBUG] [persistence.jdbc.internal.JdbcMapper] - JDBC::storeItemValue: item=GarageMandoorLHF (Type=DateTimeItem, State=2017-07-17T21:36:56)
2017-07-17 21:36:56.682 [DEBUG] [penhab.persistence.jdbc.model.ItemVO] - JDBC:ItemVO tableName=item0041; newTableName=null;
2017-07-17 21:36:56.682 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::getItemType: Try to use ItemType DATETIMEITEM for Item GarageMandoorLHF
2017-07-17 21:36:56.682 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::storeItemValueProvider: item 'GarageMandoorLHF' as Type 'DATETIMEITEM' in 'item0041' with state '2017-07-17T21:36:56'
2017-07-17 21:36:56.683 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::storeItemValueProvider: getState: '2017-07-17T21:36:56'
2017-07-17 21:36:56.683 [DEBUG] [penhab.persistence.jdbc.model.ItemVO] - JDBC:ItemVO setValueTypes dbType=TIMESTAMP(3); javaType=class java.sql.Date;
2017-07-17 21:36:56.684 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::storeItemValueProvider: DateTimeItem: '2017-07-17'
2017-07-17 21:36:56.684 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::doStoreItemValue sql=INSERT INTO item0041 (TIME, VALUE) VALUES( NOW(3), ? ) ON DUPLICATE KEY UPDATE VALUE= ? value='2017-07-17'
2017-07-17 21:36:56.686 [DEBUG] [jdbc.internal.JdbcPersistenceService] - JDBC: Stored item 'GarageMandoorLHF' as '2017-07-17T21:36:56' in SQL database at Mon Jul 17 21:36:56 EDT 2017 in 5 ms.

TLDR: It works.

I thought I had commented it out.  It is possible I didn't wait long enough.  It seems to take a little while sometimes to load addons or remove them.

I tried again and received this error:
```
2017-12-21 16:05:50.129 [ERROR] [ence.jdbc.internal.JdbcConfiguration] - JDBC::updateConfig: could NOT load JDBC-driverClassName or JDBC-dataSourceClassName. ClassNotFoundException: 'org.mariadb.jdbc.Driver cannot be found by org.openhab.persistence.jdbc_1.11.0.201712170333'
2017-12-21 16:05:50.134 [WARN ] [ence.jdbc.internal.JdbcConfiguration] -

        !!!
        To avoid this error, place an appropriate JDBC driver file for serviceName 'mariadb' in addons directory.
        Copy missing JDBC-Driver-jar to your OpenHab/addons Folder.
        !!!
        DOWNLOAD:
        MariaDB:   version >= 1.2.0 from              http://mvnrepository.com/artifact/org.mariadb.jdbc/mariadb-java-client

2017-12-21 16:05:50.137 [WARN ] [persistence.jdbc.internal.JdbcMapper] - JDBC::openConnection: no driver available!
```

I downloaded this: http://central.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/2.2.0/mariadb-java-client-2.2.0.jar

At first it didn't seem to fix it but I walked away frustrated and it finally loaded and I was able to see time being stamped into my mariadb items now.

--- a/bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/db/JdbcBaseDAO.java
+++ b/bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/db/JdbcBaseDAO.java
@@ -432,10 +432,9 @@ public class JdbcBaseDAO {
             logger.debug("JDBC::storeItemValueProvider: newVal.intValue: '{}'", newVal.intValue());
             vo.setValue(newVal.intValue());
         } else if ("DATETIMEITEM".equals(itemType)) {
-            // vo.setValueTypes(getSqlTypes().get(itemType), java.util.Date.class);
-            vo.setValueTypes(getSqlTypes().get(itemType), java.sql.Date.class);
+            vo.setValueTypes(getSqlTypes().get(itemType), java.sql.Timestamp.class);
             Calendar x = ((DateTimeType) item.getState()).getCalendar();
-            java.sql.Date d = new java.sql.Date(x.getTimeInMillis());
+            java.sql.Timestamp d = new java.sql.Timestamp(x.getTimeInMillis());
             logger.debug("JDBC::storeItemValueProvider: DateTimeItem: '{}'", d);
             vo.setValue(d);
         } else {

